app/src/main/java/de/xikolo/utils/StorageUtil.kt

```diff
@@ -0,0 +1,133 @@
+    @JvmStatic
+    fun getInternalStorage(c: Context): File = getStorages(c)[0]
+
+    /* Returns the "SD Card" storage or null if not available. */
+    @JvmStatic
+    fun getSdcardStorage(c: Context): File? {
+        val storages = getStorages(c)
+        return if (storages.size > 1) storages[1] else null
+    }
+
+    @JvmStatic
+    fun getStorage(c: Context): File {
+        val type = getStoragePreference(c)
+        when (type) {
+            StorageUtil.StorageType.SDCARD -> {
+                val sdCard = getSdcardStorage(c)
+                return sdCard ?: getInternalStorage(c)
+            }
+            StorageUtil.StorageType.INTERNAL -> return getInternalStorage(c)
+            else -> return getInternalStorage(c)
+        }
+    }
+
+    @JvmStatic
+    fun getStoragePreference(c: Context): StorageType = toStorageType(c, ApplicationPreferences().storage!!)
+
+    @JvmStatic
+    fun migrateAsync(from: File, to: File, callback: StorageMigrationCallback) {
+        Thread(Runnable {
+            copiedFilesCount = 0
+            if (from.exists()) {
+                val totalFiles = FileUtil.folderFileNumber(from)
+                callback.onProgressChanged(copiedFilesCount)
+                move(from, to, callback)
+                callback.onCompleted(copiedFilesCount == totalFiles)
+            } else
+                callback.onCompleted(false)
+        }).start()
+    }
+
+    private var copiedFilesCount = 0
+    private fun move(sourceFile: File, destFile: File, callback: StorageUtil.StorageMigrationCallback) {
+        if (sourceFile.isDirectory) {
+            for (file in sourceFile.listFiles()!!) {
+                move(file, File(destFile.absolutePath + File.separator + file.name), callback)
+            }
+        } else {
+            try {
+                destFile.mkdirs()
+                sourceFile.copyTo(destFile, true)
+                copiedFilesCount++
+                callback.onProgressChanged(copiedFilesCount)
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+        sourceFile.delete()
+    }
+
+    @JvmStatic
+    fun buildMigrationMessage(c: Context, from: StorageType): String {
+        var currentStorage = getStoragePreference(c)
+        var current = c.getString(R.string.settings_title_storage_internal)
+        if (currentStorage == StorageType.INTERNAL)
+            current = c.getString(R.string.settings_title_storage_external)
+
+        var destination = c.getString(R.string.settings_title_storage_external)
+        if (from == StorageType.SDCARD)
+            destination = c.getString(R.string.settings_title_storage_internal)
+
+        return c.getString(R.string.dialog_storage_migration, current, destination)
+    }
+
+    @JvmStatic
+    fun toStorageType(c: Context, s: String): StorageType = if (s == c.getString(R.string.settings_title_storage_external)) StorageType.SDCARD else StorageType.INTERNAL
```

Comment:
This line is too long and contains too much logic (method definition, if statement, returns), please use line breaks.

Reply:
Okay.